Dockerfile build upgrades dependencies.

### Link to reproduction

Reproduction notes below.

### To Reproduce

Our prod build suddenly started failing yesterday with `payload cms Error: Cannot find module 'functions-have-names'`

This was caused by dependency update that only occurred with the Dockerfile build.

Recommend adding: `COPY yarn.lock ./`

```
FROM node:18-alpine as base

FROM base as builder

WORKDIR /home/node
COPY package*.json ./

COPY . .
RUN yarn install
RUN yarn build

FROM base as runtime

ENV NODE_ENV=production

WORKDIR /home/node
COPY package*.json  ./
COPY yarn.lock ./

RUN yarn install --production
COPY --from=builder /home/node/dist ./dist
COPY --from=builder /home/node/build ./build

EXPOSE 3000

CMD ["node", "dist/server.js"]
```

### Describe the Bug

See above.

### Payload Version

1.11.8

--- a/templates/blank/Dockerfile
+++ b/templates/blank/Dockerfile
@@ -16,6 +16,7 @@ ENV PAYLOAD_CONFIG_PATH=dist/payload.config.js
 
 WORKDIR /home/node/app
 COPY package*.json  ./
+COPY yarn.lock ./
 
 RUN yarn install --production
 COPY --from=builder /home/node/app/dist ./dist

--- a/templates/ecommerce/Dockerfile
+++ b/templates/ecommerce/Dockerfile
@@ -16,6 +16,7 @@ ENV PAYLOAD_CONFIG_PATH=dist/payload.config.js
 
 WORKDIR /home/node/app
 COPY package*.json  ./
+COPY yarn.lock ./
 
 RUN yarn install --production
 COPY --from=builder /home/node/app/dist ./dist

--- a/templates/website/Dockerfile
+++ b/templates/website/Dockerfile
@@ -16,6 +16,7 @@ ENV PAYLOAD_CONFIG_PATH=dist/payload.config.js
 
 WORKDIR /home/node/app
 COPY package*.json  ./
+COPY yarn.lock ./
 
 RUN yarn install --production
 COPY --from=builder /home/node/app/dist ./dist

